### PR TITLE
Add Execute and Query Operations for PostgreSQL Specific Array types

### PR DESCRIPTION
**Pull request by SasinduDilshara**
## Purpose
Add Execute and Query Operations for PostgreSQL Specific Array types.
ex:- `point array`, `network array`
Resolves https://github.com/ballerina-platform/ballerina-standard-library/issues/1273

## Examples
```
postgresql:Point point1= {x:1,y:2};
postgresql:Point point2= {x:2,y:3};
PointArrayValue pointArray = [point1,point2];

sql:ParameterizedQuery sqlQuery =
        `INSERT INTO ArrayTypes (row_id,  point_array ) VALUES(${rowId},  ${pointArray})`;
```
##Related PR
https://github.com/ballerina-platform/module-ballerina-sql/pull/199

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: postgresql-ballerina/types.bal [hunk 1]
```diff
@@ -42,6 +52,18 @@ public distinct class CidrValue {
     }  
 }
 
+# Represents Cidr array PostgreSQL Field
+#
+# + value - Value of parameter passed into the SQL statement
+public distinct class CidrArrayValue {
```
Comment (kalaiyarasiganeshalingam) [inline postgresql-ballerina/types.bal:58]: @daneshk @niveathika Can we remove the suffix as `Value` from the Array type names?

```suggestion
public distinct class CidrArrayValue {
```

File: .github/workflows/publish-release.yml [hunk 2]
```diff
@@ -57,4 +59,4 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
         run: |
           curl -fsSL https://github.com/github/hub/raw/master/script/get | bash -s 2.14.1
-          bin/hub pull-request --base main -m "[Automated] Sync main after "$VERSION" release"
+          bin/hub pull-request -m "[Automated] Sync master after "$VERSION" release"
```
Comment (niveathika) [inline .github/workflows/publish-release.yml:62]: The base is main for this repo. Check in all places

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
 
 on:
- pull_request:
- push:
-   branches-ignore:
-     - main
+  pull_request:
+  push:
+    branches-ignore:
+      - master
```
Comment (niveathika) [inline .github/workflows/pull-request.yml:7]:
```suggestion
      - main
```

File: postgresql-ballerina/build.gradle [hunk 4]
```diff
@@ -17,282 +17,86 @@
...
+credentials {
+    username System.getenv('packageUser')
+    password System.getenv('packagePAT')
+}
+}
+}
+dependencies {
+    classpath "io.ballerina:plugin-gradle:${project.ballerinaGradlePluginVersion}"
+}
+}
```
Comment (niveathika) [inline postgresql-ballerina/build.gradle:33]: Format

File: postgresql-ballerina/client.bal [hunk 5]
```diff
@@ -126,103 +125,6 @@ public client class Client {
     }
 }
 
-# Provides a set of configurations for the PostgreSQL client to be passed internally within the module.
```
Comment (niveathika) [inline postgresql-ballerina/client.bal:129]: Why are these removed?

File: postgresql-ballerina/types.bal (hunk 1 above)
Comment (niveathika) [reply postgresql-ballerina/types.bal:58]: +1

File: postgresql-ballerina/client.bal (hunk 5 above)
Comment (kalaiyarasiganeshalingam) [reply postgresql-ballerina/client.bal:129]: Those are removed from this as these are moved to [record_type.bal](https://github.com/ballerina-platform/module-ballerinax-postgresql/pull/7/files#diff-00377e4abd9f8117b7b23ef8d6f98a88079e8394b7d512954f2f76212607bf97R21)

**Review by niveathika — APPROVED** [review]
LGTM

File: postgresql-ballerina/types.bal (hunk 1 above)
Comment (kalaiyarasiganeshalingam) [reply postgresql-ballerina/types.bal]: I have changed. It looks good[1] but some array types[2] defined in the SQL module with the `Value` suffix. Can we do this change for those names?

[1] https://github.com/ballerina-platform/module-ballerinax-postgresql/pull/7/files#diff-a9e5a92ee7ba7bf543bf2c119aaa7db5c7ea3b8d4a6681ce558171dfd2f1e94cR1302
[2] https://github.com/ballerina-platform/module-ballerinax-postgresql/pull/7/files#diff-a9e5a92ee7ba7bf543bf2c119aaa7db5c7ea3b8d4a6681ce558171dfd2f1e94cR1183

File: postgresql-ballerina/record_type.bal [hunk 6]
```diff
@@ -0,0 +1,338 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline postgresql-ballerina/record_type.bal:1]: shall we rename the file to `record_types.bal`

File: postgresql-ballerina/types.bal (hunk 1 above)
Comment (daneshk) [reply postgresql-ballerina/types.bal:58]: you have a valid point. we don't need the suffix `Value`, but let's keep it for consistency.

File: postgresql-native/build.gradle [hunk 7]
```diff
@@ -30,9 +30,15 @@ configurations {
...
 dependencies {
     jacocoRuntime "org.jacoco:org.jacoco.agent:${jacoco.toolVersion}:runtime"
-    compile "org.postgresql:postgresql:${postgreSQLDriverVersion}"
+
     checkstyle project(":build-config:checkstyle")
     checkstyle "com.puppycrawl.tools:checkstyle:${puppycrawlCheckstyleVersion}"
+    compile group: 'org.ballerinalang', name: 'ballerina-runtime', version: "${ballerinaLangVersion}"
+    compile group: 'org.ballerinalang', name: 'sql-native', version: "${stdlibSqlVersion}"
+    compile group: 'org.ballerinalang', name: 'time-native', version: "${stdlibTimeVersion}"
+    implementation group: 'org.postgresql', name: 'postgresql', version: "${postgreSQLDriverVersion}"
```
Comment (daneshk) [inline postgresql-native/build.gradle:39]: is this Postgres driver dependency?

File: postgresql-ballerina/params.bal [hunk 8]
```diff
@@ -0,0 +1,589 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline postgresql-ballerina/params.bal:1]: shall we rename the file to `procedure-params.bal`

File: postgresql-ballerina/record_type.bal (hunk 6 above)
Comment (daneshk) [reply postgresql-ballerina/record_type.bal:1]: shall we move config records to client.bal file. IMO it is easier to read

File: postgresql-native/build.gradle (hunk 7 above)
Comment (kalaiyarasiganeshalingam) [reply postgresql-native/build.gradle:39]: Yes, It is dependency.
https://github.com/ballerina-platform/module-ballerinax-postgresql/blob/main/postgresql-native/src/main/java/module-info.java#L24

**Review by daneshk — APPROVED** [review]
